client/components/EventForm/index.jsx

```diff
@@ -209,66 +209,25 @@ export class Component extends React.Component {
             readOnly: !recurringRulesEditable,
         }
 
-        const eventActions = {
-            'EVENT_HISTORY': {
-                label: 'View Event History',
-                callback: this.viewEventHistory.bind(this),
-            },
-            'UNSPIKE_EVENT': {
-                label: 'Unspike Event',
-                callback: unspikeEvent.bind(null, initialValues),
-            },
-            'CREATE_PLANNING': {
-                label: 'Create Planning Item',
-                callback: () => addEventToCurrentAgenda(initialValues),
-            },
-            'DUPLICATE_EVENT': {
-                label: 'Duplicate Event',
-                callback: () => duplicateEvent(initialValues),
-            },
-            'SPIKE_EVENT': {
-                label: 'Spike Event',
-                callback: () => spikeEvent(initialValues),
-            },
-        }
         let itemActions = []
-
-        const populateItemActions = () => {
-            itemActions.unshift(eventActions.EVENT_HISTORY)
-
-            if (eventUtils.canCreatePlanningFromEvent(initialValues, session, privileges)) {
-                itemActions.unshift(eventActions.CREATE_PLANNING)
+        if (existingEvent) {
+            const callBacks = {
+                [GENERIC_ITEM_ACTIONS.SPIKE.label]: () => spikeEvent(initialValues),
```

Comment:
Should we bind this (as well as duplicateEvent and addEventToCurrentAgenda) instead of calling it from an arrow function?